File: LeetCode/Medium/0056-merge-intervals/0056-merge-intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        vector<vector<int>> ret;
        sort(intervals.begin(), intervals.end()); 
        vector<int> temp{intervals[0][0], intervals[0][1]};

        for (int i = 1; i < intervals.size(); ++i){
            if (temp[1] >= intervals[i][0]) temp[1] = max(temp[1], intervals[i][1]);
            else {
                ret.push_back(temp);
                temp = intervals[i];
            }
        }
        
        ret.push_back(temp);

        return ret;
    }
};

File: LeetCode/Medium/0056-merge-intervals/0056-merge-intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        n = len(intervals)
        ret = []
        intervals.sort(key=lambda x: x[0])
        i = 0
        
        while i < n:
            temp = [intervals[i][0], intervals[i][1]]

            i += 1
            while i  < n and temp[1] >= intervals[i][0]:
                temp[1] = max(temp[1], intervals[i][1])
                i += 1

            ret.append(temp)

        return ret